packages/material-ui/src/ListItemButton/ListItemButton.js

```diff
@@ -0,0 +1,259 @@
+      backgroundColor: 'transparent',
+    },
+  },
+  [`&.${listItemButtonClasses.selected}`]: {
+    backgroundColor: alpha(theme.palette.primary.main, theme.palette.action.selectedOpacity),
+    [`&.${listItemButtonClasses.focusVisible}`]: {
+      backgroundColor: alpha(
+        theme.palette.primary.main,
+        theme.palette.action.selectedOpacity + theme.palette.action.focusOpacity,
+      ),
+    },
+  },
+  [`&.${listItemButtonClasses.selected}:hover`]: {
+    backgroundColor: alpha(
+      theme.palette.primary.main,
+      theme.palette.action.selectedOpacity + theme.palette.action.hoverOpacity,
+    ),
+    // Reset on touch devices, it doesn't add specificity
+    '@media (hover: none)': {
+      backgroundColor: alpha(theme.palette.primary.main, theme.palette.action.selectedOpacity),
+    },
+  },
+  [`&.${listItemButtonClasses.focusVisible}`]: {
+    backgroundColor: theme.palette.action.focus,
+  },
+  [`&.${listItemButtonClasses.disabled}`]: {
+    opacity: theme.palette.action.disabledOpacity,
+  },
+  /* Styles applied to the root element if `divider={true}`. */
+  ...(styleProps.divider && {
+    borderBottom: `1px solid ${theme.palette.divider}`,
+    backgroundClip: 'padding-box',
+  }),
+  /* Styles applied to the root element if `alignItems="flex-start"`. */
+  ...(styleProps.alignItems === 'flex-start' && {
+    alignItems: 'flex-start',
+  }),
+  /* Styles applied to the root element unless `disableGutters={true}`. */
+  ...(!styleProps.disableGutters && {
+    paddingLeft: 16,
+    paddingRight: 16,
+  }),
+  /* Styles applied to the root element if dense. */
+  ...(styleProps.dense && {
+    paddingTop: 4,
+    paddingBottom: 4,
+  }),
+}));
+
+const ListItemButton = React.forwardRef(function ListItemButton(inProps, ref) {
+  const props = useThemeProps({ props: inProps, name: 'MuiListItemButton' });
+  const {
+    autoFocus = false,
+    component = 'div',
+    children,
+    alignItems = 'center',
+    dense = false,
+    disableGutters = false,
+    divider = false,
+    selected = false,
+    focusVisibleClassName,
+    ...other
+  } = props;
+
+  const context = React.useContext(ListContext);
+  const childContext = {
+    dense: dense || context.dense || false,
+    alignItems,
+    disableGutters,
+  };
+
+  const listItemRef = React.useRef(null);
+  useEnhancedEffect(() => {
+    if (autoFocus) {
+      if (listItemRef.current) {
+        listItemRef.current.focus();
+      } else if (process.env.NODE_ENV !== 'production') {
+        console.error(
+          'Material-UI: Unable to set focus to a ListItem whose component has not been rendered.',
+        );
+      }
+    }
+  }, [autoFocus]);
+
+  const styleProps = {
+    ...props,
+    alignItems,
+    dense: childContext.dense,
+    disableGutters,
+    divider,
+    selected,
+  };
+
+  const classes = useUtilityClasses(styleProps);
+
+  const handleRef = useForkRef(listItemRef, ref);
+
+  return (
+    <ListContext.Provider value={childContext}>
```

Comment:
Would it make more sense for only ListItem to have this?

Reply:
I think it is needed, otherwise developer might confused.

```jsx
<Box component="nav">
  <ListItemButton dense>
    <ListItemText>Hello</ListItemText>
    <KeyboardArrowRight />
  </ListItemButton>
</Box>
```